src/app/ApplicationTemplate.Business/KillSwitch/KillSwitchService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Reactive.Linq;
+using ApplicationTemplate.DataAccess;
+
+namespace ApplicationTemplate.Business;
+
+/// <summary>
+/// Implementation of the IKillSwitchService.
+/// </summary>
+public class KillSwitchService : IKillSwitchService, IDisposable
+{
+	private readonly IKillSwitchRepository _killSwitchRepository;
+	private IDisposable _subscription;
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="KillSwitchService"/> class.
+	/// </summary>
+	/// <param name="killSwitchRepository">The <see cref="IKillSwitchRepository"/>.</param>
+	public KillSwitchService(IKillSwitchRepository killSwitchRepository)
+	{
+		_killSwitchRepository = killSwitchRepository;
+
+		bool isMock = _killSwitchRepository is KillSwitchRepositoryMock;
+
+		_subscription = _killSwitchRepository.KillSwitchActivatedObservable.DistinctUntilChanged().Subscribe(isActivated =>
+		{
+			KillSwitchActivationChanged?.Invoke(this, new KillSwitchEventArgs(isActivated));
```

Comment:
Add logs.

Reply:
I didnt add log at this place because, I'm not really sure if there's anything to log there anymore, but I added some logs in the CoreStartup

Reply:
The business service should log an information-level message when the kill switch activation changes.